lib/text/ttml_text_parser.js

```diff
@@ -49,45 +49,15 @@ shaka.text.TtmlTextParser = class {
     const ttsNs = TtmlTextParser.styleNs_;
     const str = shaka.util.StringUtils.fromUTF8(data);
     const cues = [];
-    const parser = new DOMParser();
-    let xml = null;
 
     // dont try to parse empty string as
     // DOMParser will not throw error but return an errored xml
     if (str == '') {
       return cues;
     }
 
-    try {
-      xml = parser.parseFromString(str, 'text/xml');
-    } catch (exception) {
-      throw new shaka.util.Error(
-          shaka.util.Error.Severity.CRITICAL,
-          shaka.util.Error.Category.TEXT,
-          shaka.util.Error.Code.INVALID_XML,
-          'Failed to parse TTML.');
-    }
-
-    if (xml) {
-      const parserError = xml.getElementsByTagName('parsererror')[0];
-      if (parserError) {
-        throw new shaka.util.Error(
-            shaka.util.Error.Severity.CRITICAL,
-            shaka.util.Error.Category.TEXT,
-            shaka.util.Error.Code.INVALID_XML,
-            parserError.textContent);
-      }
-
-      const tt = xml.getElementsByTagName('tt')[0];
-      // TTML should always have tt element.
-      if (!tt) {
-        throw new shaka.util.Error(
-            shaka.util.Error.Severity.CRITICAL,
-            shaka.util.Error.Category.TEXT,
-            shaka.util.Error.Code.INVALID_XML,
-            'TTML does not contain <tt> tag.');
-      }
-
+    const tt = XmlUtils.parseXmlString(str, 'tt');
+    if (tt) {
```

Comment:
I think this would be more readable if this checked `if (!tt)` and, if so, threw the error here. That puts the error handling right next to where the error is produced.
That would also mean that the rest of the function below this could be indented one less.

Reply:
Agreed, fixed.